src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -35,9 +39,79 @@ public LibGpiodDriver()
             _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
         }
 
+        private SafeLineHandle SubscribeForEvent(int pinNumber, PinEventTypes eventType)
+        {
+            int eventSuccess = -1; 
+            SafeLineHandle pinHandle = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pinHandle != null)
+            {
+                if (eventType.HasFlag(PinEventTypes.Rising) || eventType.HasFlag(PinEventTypes.Falling))
+                {
+                    eventSuccess = Interop.RequestBothEdgeEventForLine(pinHandle, $"Listen {pinNumber} for both edge event");
+                }
+                else
+                {
+                    throw ExceptionHelper.GetArgumentException(ExceptionResource.InvalidEventType);
+                }
+            }
+
+            if (eventSuccess < 0)
+            {
+                throw ExceptionHelper.GetIOException(ExceptionResource.RequestEventError, pinNumber, Marshal.GetLastWin32Error());
+            }
+            return pinHandle;
+        }
+
+        private Task InitializeEventDetectionTask(LibGpiodDriverEventHandler eventHandler, CancellationToken token)
+        {
+            Task pinListeningThread = Task.Run(() =>
```

Comment:
you should also pass cancellation token to the Task.Run - this will handle the case where event is already cancelled before task event starts and prevent the task from starting